sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSensor.java

```diff
@@ -121,15 +130,25 @@ public void execute(SensorContext context) {
     if (pythonVersionParameter.length != 0){
       ProjectPythonVersion.setCurrentVersions(PythonVersionUtils.fromStringArray(pythonVersionParameter));
     }
+    setTelemetry(context, pythonVersionParameter);
     CacheContext cacheContext = CacheContextImpl.of(context);
     PythonIndexer pythonIndexer = this.indexer != null ? this.indexer : new SonarQubePythonIndexer(pythonFiles, cacheContext, context);
     pythonIndexer.setSonarLintCache(sonarLintCache);
     TypeShed.setProjectLevelSymbolTable(pythonIndexer.projectLevelSymbolTable());
     PythonScanner scanner = new PythonScanner(context, checks, fileLinesContextFactory, noSonarFilter, PythonParser.create(), pythonIndexer);
     scanner.execute(pythonFiles, context);
+    sensorTelemetryStorage.send(context);
     durationReport.stop();
   }
 
+  private void setTelemetry(SensorContext context, String[] pythonVersionParameter) {
+    var isVersionSet = pythonVersionParameter.length != 0 || context.runtime().getProduct() == SonarProduct.SONARLINT;
+    if (pythonVersionParameter.length != 0) {
+      sensorTelemetryStorage.updateMetric(TelemetryMetricKey.PYTHON_VERSION_KEY, String.join(",", pythonVersionParameter));
```

Comment:
So that means we can set more than just booleans for the telemetry?

Reply:
Yes, CFamily is sending strings.